angular/src/app/components/compare/compare.component.ts

```diff
@@ -1,130 +1,206 @@
-      element.values.forEach(val => {
-        val.checked = (val.value === element.ref);
-      });
-    });
-
-  }
-
-  logNode(node: Node<ENVCONFIG>) {
+  logNode(node: Node<ENVCONFIG>): void {
     console.log(node);
   }
 
-  onAddEnv(event) {
-    if (!this.selectedEnvs.includes(this.selectedValue) && (this.selectedValue != "")){
+  onAddEnv(event): void {
+    if (!this.selectedEnvs.includes(this.selectedValue) && (this.selectedValue != "")) {
       this.selectedEnvs.push(this.selectedValue);
     }
-  this.generateShowlists();
-  this.getServerConfiguration();
- 
+    this.generateShowlists();
+    this.getServerConfiguration();
+    this.saveCurrentSollValues();
+
 
   }
-  getServerConfiguration() {
+  getServerConfiguration(): void {
     let envs = ServerConfiguration.ENV_LIST;
     if (this.selectedEnvs)
       envs = this.selectedEnvs;
-    this.configItemService.getServerConfiguration(envs).subscribe(data => {
+    this.elasticService.getHostinformationByNames(envs).subscribe(data => {
       let updateTime = this.configItemService.getUpdateTime(data);
-      console.log("lastclock: "+updateTime);
+      console.log("lastclock: " + updateTime);
       let temp = new Date();
-      temp.setTime(updateTime *1000 );
-      this.lastUpdate= temp;
-      console.log("lastUpdate: "+this.lastUpdate);
-      this.configItemService.itemlist = this.configItemService.createServerConf(data);
-      this.configItemService.generateTree(this.selectedEnvs);
+      temp.setTime(updateTime * 1000);
+      this.lastUpdate = temp;
+      console.log("lastUpdate: " + this.lastUpdate);
+      this.configItemService.createServerConf();
+      this.generateTree();
+
     });
   }
 
-  setSelectedValue(event) {
+  setSelectedValue(event): void {
     this.selectedValue = event.value;
   }
 
-  onDeleteEnv(event, env) {
+  onDeleteEnv(event, env): void {
     this.selectedEnvs = this.selectedEnvs.filter(e => e !== env);
     this.generateShowlists();
     this.ref.detectChanges();
     this.configItemService.generateTree(this.selectedEnvs);
 
   }
-}
 
+  saveCurrentSollValue(): void {
+    console.log("saveCurrentSollValue")
+    let envs = ServerConfiguration.ENV_LIST;
+    if (this.selectedEnvs) {
+      envs = this.selectedEnvs;
+      this.configItemService.saveAllSollValue(envs);
+    }
+  }
+
+  saveSollValue(hostName: string, key: string, istValue: string, sollValue: string): void {
+    if (this.selectedEnvs) {
+      this.elasticService.saveSollwerte(hostName, key, sollValue).subscribe(data => {
+        console.log(data);
+        let confItem: ConfigurationItem = this.configItemService.createItem(hostName, key, istValue, sollValue)
+        console.log(confItem)
+        this.configItemService.addConfigItem(confItem);
+        this.refreshItemlist();
+        this.generateTree();
+
+      });
+    }
+    this.elasticService.saveHistorySollWerte(hostName, key, sollValue).subscribe(data => {
+      console.log(data);
```

Comment:
wird damit was gemacht?

Reply:
ja, die neue Sollwerte werden damit in history -Indizes  gespeichert

Reply:
Die Frage: wird mit dem Rückgabewert etwas gemacht?